loadgen/CMakeLists.txt

```diff
@@ -11,6 +11,10 @@ message("mlperf_loadgen v${mlperf_loadgen_VERSION_MAJOR}.${mlperf_loadgen_VERSIO
 if (NOT MSVC)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -W -Wall")
 endif()
+# Extra build options can be specified by setting the MLPERF_LOADGEN_CXX_FLAGS variable
+if (MLPERF_LOADGEN_CXX_FLAGS)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${MLPERF_LOADGEN_CXX_FLAGS}")
+endif()
 message(STATUS "Using C++ compiler flags: ${CMAKE_CXX_FLAGS}")
```

Comment:
Why can't you just do `cmake -DCMAKE_CXX_FLAGS=your_flags ...` to do this?

Reply:
@nvpohanh , you can certainly use `cmake -DCMAKE_CXX_FLAGS=your_flags ...` from the command line. But, when using inference/loadgen as a within your cmake build as an subdirectory (using cmake's add_subdirectory, ExternalProject, or FetchContent) it is difficult to specify extra CXX_FLAGS since CMAKE_CXX_FLAGS cannot be scoped to just the loadgen.

For example, a use case would be to be able to write the following

```
include(FetchContent)
FetchContent_Declare(mlperf
   GIT_REPOSITORY https://github.com/mlcommons/inference.git)
set(MLPERF_LOADGEN_CXX_FLAGS "my flags")
FetchContent_MakeAvailable(mlperf)
```

to add inference/loadgen into your project with one's own CXX_FLAGS